backend/courses/management/commands/dump_redis_data.py

```diff
@@ -0,0 +1,192 @@
+from review.annotations import annotate_average_and_recent
+from review.util import get_average_and_recent_dict_single
+
+
+course_filters_pcr_allow_xlist = ~Q(title="") | ~Q(description="") | Q(sections__has_reviews=True)
+course_filters_pcr = Q(primary_listing_id=F("id")) & course_filters_pcr_allow_xlist
+course_filters_pcr_allow_xlist = ~Q(title="") | ~Q(description="") | Q(sections__has_reviews=True)
+
+section_filters_pcr = Q(course__primary_listing_id=F("course_id")) & (
+    Q(has_reviews=True)
+    | ((~Q(course__title="") | ~Q(course__description="")) & ~Q(activity="REC") & ~Q(status="X"))
+)
+
+review_filters_pcr = Q(section__course__primary_listing_id=F("section__course_id"))
+
+def get_department_objs():
+    departments = Department.objects.all()
+    for department in departments:
+        yield {
+            "code": department.code,
+            "name": department.name
+        }
+
+def get_course_objs():
+    topics = (
+        Topic.objects.filter(most_recent__semester="2022C")[:100]
+        .select_related("most_recent")
+        .prefetch_related("most_recent__primary_listing__listing_set__sections__instructors")
+    )
+    for topic in topics:
+        course = topic.most_recent
+        crosslistings = ", ".join([c.full_code for c in course.crosslistings])
+        instructors = ", ".join({instructor.name for s in course.sections.all() for instructor in s.instructors.all()})
+        course_qs = Course.objects.filter(pk=course.pk)
+        course_qs = annotate_average_and_recent(
+            course_qs,
+            match_review_on=Q(section__course__topic=topic) & review_filters_pcr,
+            match_section_on=Q(course__topic=topic) & section_filters_pcr,
+            extra_metrics=True,
+        )
+        reviewed_course = (
+            get_average_and_recent_dict_single(dict(course_qs.values()[0])) if course_qs else None
+        )
+        avg_reviews = reviewed_course["average_reviews"]
+        yield {
+            "code": course.full_code.replace("-", " "),
+            "title": course.title,
+            "crosslistings": crosslistings,
+            "instructors": instructors,
+            "description": course.description,
+            "semester": course.semester,
+            "course_quality": avg_reviews["rCourseQuality"]
+            if "rCourseQuality" in avg_reviews
+            else None,
+            "difficulty": avg_reviews["rDifficulty"] if "rDifficulty" in avg_reviews else None,
+            "work_required": avg_reviews["rWorkRequired"]
+            if "rWorkRequired" in avg_reviews
+            else None,
+        }
+
+def get_instructor_objs():
+    instructors = (
+        Instructor.objects.filter(
+            id__in=Subquery(Section.objects.filter(section_filters_pcr).values("instructors__id"))
+        )
+        .distinct()[:100]
+        .values("name", "id", "section__course__department__code")
+    )
+    instructor_set = {}
+    for inst in instructors:
```

Comment:
We should replace this entire part with this solution using django's postgres `ArrayAgg`: https://stackoverflow.com/questions/12031563/output-of-values-on-a-queryset-with-manytomany-fields

We should also note that this is a postgres specific feature and cannot be used with an SQlite backend somewhere.

Reply:
Perhaps we can leave this as an issue for future iterations? I'd imagine the performance of the dump script is less of a concern than getting this off the ground

Reply:
Yeah fair.